src/Control/Distributed/Spark/RDD.hs

```diff
@@ -112,6 +163,17 @@ textFile sc path = do
   jpath <- reflect (Text.pack path)
   call sc "textFile" [coerce jpath]
 
+-- recordLength = number of bytes
+binaryRecords :: SparkContext -> FilePath -> Int32 -> IO (RDD ByteString)
+binaryRecords sc fp recordLength = do
+  jpath <- reflect (Text.pack fp)
+  call sc "binaryRecords" [coerce jpath, coerce recordLength]
+
+-- | Reinterprets an @RDD a@ as an @RDD b@ provided that both types have
+-- the same representation on the java side.
+rddCoerce :: (Interp a ~ Interp b) => RDD a -> RDD b
```

Comment:
That could work. But doesn't `Data.Coerce.coerce` allow to coerce any pointer to a java object indiscriminately? This call would only allow to do that when the representations in java are the same.